src/core/unit.c

```diff
@@ -1085,6 +1086,7 @@ static void unit_merge_dependencies(Unit *u, Unit *other) {
                                 /* This is a dependency pointing back to the unit we want to merge with?
                                  * Suppress it (but warn) */
                                 unit_maybe_warn_about_dependency(u, other->id, UNIT_DEPENDENCY_FROM_PTR(dt));
+                                hashmap_remove(other_deps, back);
```

Comment:
hmm, now you are changing the hashmap you are iterating through. It's something I so far avoided